ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/QueryFilterNode.java

```diff
@@ -358,6 +363,27 @@ public Void visitComparisonExpression(
       }
     }
 
+    @Override
+    public Void visitLikePredicate(final LikePredicate node, final Object context) {
+      if (node.getValue() instanceof UnqualifiedColumnReferenceExp) {
+        final UnqualifiedColumnReferenceExp column = (UnqualifiedColumnReferenceExp) node.getValue();
+        final ColumnName columnName = column.getColumnName();
+        final Column col = schema.findColumn(columnName)
+                .orElseThrow(() -> invalidWhereClauseException(
+                        "Like condition on non-existent column " + columnName, isWindowed));
+        final Expression pattern = node.getPattern();
+        if (!(pattern instanceof StringLiteral || pattern instanceof NullLiteral)) {
+          throw invalidWhereClauseException(
+                  "Like condition on non-string pattern " + pattern.getClass().getName(),
+                  isWindowed);
+        }
+      } else {
+        setTableScanOrElseThrow(() -> invalidWhereClauseException("Like condition must directly "
```

Comment:
It occurs to me, even if they reference a key column, we're still going to do a table scan for this, so might want to update the messaging.

Reply:
Message changed.